memo_app.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'cgi'
+require 'json'
+
+set :environment, :production
+
+class MemoData
+  attr_accessor :memos, :id
+
+  def initialize
+    @memos = {}
+    @id = 0
+  end
+
+  def add_memo(title, content)
+    @memos[@id] = { title: title, content: content }
+    rewrite_file
+    @id += 1
```

Comment:
`@id` を止めたので、ここで id の最大値を得るようにしてみます。

今回のデータ構造だとハッシュのキーが ID になっていますね。元々の処理では、ハッシュの最後の ID を `@id` として保持していました。
ここで代わりに `Hash#keys` でキーを取得してみます。
`Hash#keys` の戻り値は配列ですね。
https://docs.ruby-lang.org/ja/latest/method/Hash/i/keys.html

`Array#map(&:to_i)` で、配列内の値をすべて数値に変換します。

その後に `Array#max` で配列内の数値で最大のものを取得します。
データが空のときに `Array#max` を呼ぶと nil が返ります。
https://docs.ruby-lang.org/ja/latest/method/Array/i/max.html

ここで `#to_i` を呼んでおくと nil を 0 に変換できるので、 `+ 1` の計算ができるようになります。
```suggestion
    id = @memos.keys.map(&:to_i).max.to_i + 1
    @memos[id] = { title: title, content: content }
    rewrite_file
```